tests/unit/mesh/test_ncmesh.cpp

```diff
@@ -297,6 +297,228 @@ TEST_CASE("pNCMesh PA diagonal",  "[Parallel], [NCMesh]")
+   // global L2 projected solution should match exactly on each.
+   Array<int> refines, serial_refines(1);
+   if (Mpi::WorldRank() == refining_rank)
+   {
+      refines.Append(0);
+   }
+
+   // Must be called on all ranks as it uses MPI calls internally.
+   // All ranks will use the global element number dictated by rank 0 though.
+   serial_refines[0] = pmesh.GetGlobalElementNum(0);
+   MPI_Bcast(&serial_refines[0], 1, MPI_INT, refining_rank, MPI_COMM_WORLD);
+
+   // Rank 0 refines the parallel mesh, all ranks refine the serial mesh
+   smesh.GeneralRefinement(serial_refines, 1); // nonconformal
+   pmesh.GeneralRefinement(refines, 1); // nonconformal
+
+   REQUIRE(pmesh.GetGlobalNE() == 8 + 1);
+   REQUIRE(smesh.GetNE() == 8 + 1);
+
+   // This is the set of (i,j) that result in face-edge constraints.
+   // These pairs were arrived at by looping over all possible i,j pairs and
+   // checking for the addition of a face-edge constraint.
+   std::vector<std::pair<int,int>> indices{{2,13}, {3,13}, {6,2}, {6,3}};
+
+   // Rank 0 has all but one element.
+   for (const auto &ij : indices)
+   {
+      int i = ij.first;
+      int j = ij.second;
+      if (Mpi::WorldRank() == refining_rank)
+      {
+         refines[0] = i;
+      }
+      // Inform all ranks of the serial mesh
+      serial_refines[0] = pmesh.GetGlobalElementNum(i);
+      MPI_Bcast(&serial_refines[0], 1, MPI_INT, 0, MPI_COMM_WORLD);
+
+      ParMesh tmp(pmesh);
+      tmp.GeneralRefinement(refines);
+
+      REQUIRE(tmp.GetGlobalNE() == 1 + 8 - 1 + 8); // 16 elements
+
+      Mesh stmp(smesh);
+      stmp.GeneralRefinement(serial_refines);
+      REQUIRE(stmp.GetNE() == 1 + 8 - 1 + 8); // 16 elements
+
+      if (Mpi::WorldRank() == refining_rank)
+      {
+         refines[0] = j;
+      }
+      // Inform all ranks of the serial mesh
+      serial_refines[0] = tmp.GetGlobalElementNum(j);
+      MPI_Bcast(&serial_refines[0], 1, MPI_INT, 0, MPI_COMM_WORLD);
+
+      ParMesh ttmp(tmp);
+      ttmp.GeneralRefinement(refines);
+
+      REQUIRE(ttmp.GetGlobalNE() == 1 + 8 - 1 + 8 - 1 + 8); // 23 elements
+
+      Mesh sttmp(stmp);
+      sttmp.GeneralRefinement(serial_refines);
+      REQUIRE(sttmp.GetNE() == 1 + 8 - 1 + 8 - 1 + 8); // 23 elements
+
+      // Loop over interior faces, fill and check face transform on the serial.
+      for (int iface = 0; iface < sttmp.GetNumFaces(); ++iface)
+      {
+         const auto face_transform = sttmp.GetFaceElementTransformations(iface);
+
+         CHECK(std::abs(face_transform->CheckConsistency(0)) < 1e-12);
+      }
+
+      for (int iface = 0; iface < ttmp.GetNumFacesWithGhost(); ++iface)
+      {
+         const auto face_transform = ttmp.GetFaceElementTransformations(iface);
+
+         CHECK(std::abs(face_transform->CheckConsistency(0)) < 1e-12);
+      }
+
+      // Use P4 to ensure there's a few fully interior DOF.
+      check_l2_projection(ttmp, sttmp, 4);
+
+      ttmp.ExchangeFaceNbrData(); // This causes issues.
```

Comment:
Is this comment obsolete? Doesn't this PR resolve the issues?

Reply:
Resolved in https://github.com/mfem/mfem/pull/3587/commits/c790cf942bb329d9b90a6ca1de6b31fccbc1d7d9